File: app/src/main/java/me/aluceps/sandbox/repository/ConnpassRepository.java
package me.aluceps.sandbox.repository;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Single;
import me.aluceps.sandbox.api.ConnpassClient;
import me.aluceps.sandbox.model.ConnpassEvent;

@Singleton
public class ConnpassRepository implements ConnpassRemoteDataSource {

    private final ConnpassClient client;

    @Inject
    public ConnpassRepository(ConnpassClient client) {
        this.client = client;
    }

    @Override
    public Single<ConnpassEvent> events() {
        return client.events();
    }
}

File: app/src/main/java/me/aluceps/sandbox/repository/ConnpassRepository.kt
package me.aluceps.sandbox.repository

import io.reactivex.Single
import me.aluceps.sandbox.api.ConnpassClient
import me.aluceps.sandbox.model.ConnpassEvent
import me.aluceps.sandbox.model.RequestParams
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ConnpassRepository @Inject
constructor(private val client: ConnpassClient) : ConnpassRemoteDataSource {

    override fun events(params: RequestParams): Single<ConnpassEvent> = client.events(params)
}
